src/pages/generate.tsx

```diff
@@ -87,10 +75,36 @@ export default function ClientIdentifierGenerator() {
     defaultMaxAge: undefined,
   };
 
+  /** Helper to load form state and values from storage or defaults. */
+  const getSessionStorageStateOrDefault: () => {
+    initialValues: FormParameters;
+    initialStates: ReturnType<typeof getEmptyFormState>;
+  } = () => {
+    // Load state from session store, if present, else use empty values.
+    const storedFormValues = sessionStorage.getItem("formValues");
+    const formValidationState = sessionStorage.getItem("formValidationState");
+    const initialValues = storedFormValues
+      ? JSON.parse(storedFormValues)
+      : initialFormValues;
+    const initialStates =
+      formValidationState && storedFormValues
+        ? JSON.parse(formValidationState)
+        : getEmptyFormState();
```

Comment:
`JSON.parse` can throw, make sure you do this safely.

Reply:
Addressed in https://github.com/inrupt/client-identifier-helper/pull/185/commits/ec0e24242114d647bbf77652364bc7891793ae7e.